Overhaul trustee docs

fitzthum opened the pull request:
Well it's time to bring the attestation section up to the standard of the rest of the docs.

I am removing some older stuff that was copied from Github and is more intended for developers. These docs are for users.

In the future I hope we can bring some of that information back to the website (maybe in a section on advanced configurations), but for now let's focus on making it simple.

There are probably mistakes in here and things that could be more rigorous, but let's just get something up there and then try to iterate (unless you have some specific fixes to suggest on this PR).

cc @mythi @mkulke @lmilleri

mkulke reviewed (commented):
Would it make sense to move the "KBS backed by AKV" entry to the "Examples" section? (and maybe spell out "Azure Key Vault")

fitzthum commented:
> Would it make sense to move the "KBS backed by AKV" entry to the "Examples" section? (and maybe spell out "Azure Key Vault")

It seems to fit pretty well in the resource section but I can move it to examples if you would prefer that. Changing the name is no problem.

mkulke commented:
> > Would it make sense to move the "KBS backed by AKV" entry to the "Examples" section? (and maybe spell out "Azure Key Vault")
> 
> It seems to fit pretty well in the resource section but I can move it to examples if you would prefer that. Changing the name is no problem.

I don't have a strong opinion either. ideally, we would use tags consistently, so the hierarchies do not matter that much for discoverability

mythi commented on content/en/docs/attestation/installation/docker.md:
```diff
@@ -0,0 +1,39 @@
+
+### Uninstall
+
+Stop docker compose.
```
```suggestion
Stop Trustee.
```

mythi commented on content/en/docs/attestation/client-tool/_index.md:
````diff
@@ -1,16 +1,75 @@
+sudo make install cli
+```
+
+The `sample_only` feature is used to avoid building hardware attesters into the KBS Client.
+In addition to configuring the KBS, the client can also be used inside of an enclave
+to retrieve resources.
+In this case the attester for that enclave must be built.
+You can remove the `sample_only` feature to build all the attesters.
+This will require extra dependencies.
````
The comments about attesters could be relevant for the ORAS based setup too so perhaps some of this could be moved to the intro. The ORAS section could also mention that the pre-built `kbs-client` has no attester support either.

bpradipt replied:
+1.  It's important to clarify that the published kbs-client has only sample attester support.
May be as a next step we can create a package with all coco tools in one place to make it easier for users to try out without building the tools.

I'm doing something like this here - https://github.com/confidential-devhub/coco-tools/blob/main/Containerfile

fitzthum replied:
I added some more info about the feature to the intro section to clarify. This does reveal that our tools aren't quite as easy to consume as they should be.

bpradipt replied:
> This does reveal that our tools aren't quite as easy to consume as they should be.
Yes.

mythi replied:
> This does reveal that our tools aren't quite as easy to consume as they should be.

FWIW, I'm planning to re-work `tdx-attester`  to be fully Rust based by default.

mythi commented on content/en/docs/attestation/policies/_index.md:
```diff
@@ -0,0 +1,200 @@
+Init-data allows dynamic, measured configurations to be propagated to a guest.
+Those configurations can also be provided to Trustee as part of the KBS protocol.
+If they are, policies can check the init-data, which is located at `submods.cpu.ear.veraison.annotated-evidence.init_data_claims`.
+This is only supported on platforms that have a hardware init-data field (such as `HostData` on SEV-SNP).
```
```suggestion
This is only supported on platforms that have a hardware init-data field (such as `HostData` on SEV-SNP or `mrconfigid/mrowner/mrownerconfig` on TDX).
```

mythi reviewed (commented):
This is great, especially the details on dealing with the policies! I made some minor suggestions and added a comment on the `kbs-client` wording on the attesters support.

bpradipt reviewed (approved):
Just a minor nit on the kbs-client. 
This is a really good addition. Especially the policy pieces

fitzthum commented:
> don't have a strong opinion either. ideally, we would use tags consistently, so the hierarchies do not matter that much for discoverability

Ok, I changed the name and added a link from the AZ example but kept the file in the same place. Both the example and this page have the `azure` tag but maybe we can figure out more tags in the future. Feel free to change or move in the future if you would like.
